src/Main.elm

```diff
@@ -1,10 +1,197 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser
+import Data.Audience as Aud
+import Data.AudienceFolder as AudFolders
+import Debug exposing (toString)
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick)
 
 
-main : Html msg
+main : Program () Model Msg
 main =
-    Html.text "There will be app soon!"
+    Browser.sandbox
+        { init = init
+        , update = update
+        , view = view
+        }
+
+
+type alias Model =
+    { folders : List AudFolders.AudienceFolder
+    , opened : Opened
+    , audience : List Aud.Audience
+    , breadcrumbs : Breadcrumbs
+    }
+
+
+type alias Opened =
+    { currentId : Maybe Int
+    , parentId : Maybe Int
+    , usedIdList : List Int
+    , parentName : List String
+    , state : Bool
+    }
+
+
+type alias Breadcrumbs =
+    { breadCrumbName : List String
+    , breadCrumbId : List Int
+    }
+
+
+type Msg
+    = MsgFolderOpened Int String
+    | MsgFolderClosed
+
+
+init : Model
+init =
+    initModel
+
+
+initModel : Model
+initModel =
+    { folders = AudFolders.audFolders
+    , opened = { currentId = Nothing, parentId = Nothing, usedIdList = [], parentName = [], state = False }
+    , audience = Aud.audience
+    , breadcrumbs = { breadCrumbName = [], breadCrumbId = [] }
+    }
+
+
+update : Msg -> Model -> Model
+update msg model =
+    let
+        newOpened =
+            model.opened
+
+        newBreadcrumb =
+            model.breadcrumbs
+
+        newBreadcrumbId id =
+            id :: newBreadcrumb.breadCrumbId
+
+        newParentId =
+            newOpened.currentId
+
+        newCurrentId =
+            newOpened.parentId
+
+        newParentName name =
+            name :: newOpened.parentName
+
+        newUsedIdList id =
+            id :: newOpened.usedIdList
+
+        newBreadcrumbpName name =
+            name :: newBreadcrumb.breadCrumbName
```

Comment:
Although the cons operator (`::`) shows that `Breadcrumbs.breadCrumbName` is a `List`, it could be confusing in a larger codebase. `newBreadcrumb.breadCrumbName` alone can lead to think that it is a single value, and, although we have the typesystem to save us from this sort of things, it is always convinient to make it as easy-viewing as possible for you in the future.